### PR TITLE
Post Announcements CLI

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

This PR implements CLIVE (short for CLI Live) that allows the user to authenticate with Auth0 prior to making BFF requests from the command line. It also implements a CLI command for making network announcements. The basic flow is as follows:

```
$ bff login 
$ bff announce "my announcement to the TRISA network"
```

CLIVE works using 3 legged OAuth without requiring an access token. First CLIVE generates a `code_verifier` (random 32 bytes of data) then creates a `code_challenge` -- a SHA256 hash of the verifier. CLIVE then creates a URL with the code challenge, the client ID of the Auth0 app that I created for this (configuration is in Auth0 and undocumented), and opens a browser window with the link. At the same time it starts a lightweight http server on localhost.

Auth0 redirects the request in the browser to the callback url at the localhost, which is handled by the CLIVE server. CLIVE fetches the tokens in the request and then POSTs the verifier to the auth0 endpoint in the backend. If everything checks out, Auth0 gives CLIVE an access token, which it saves to disk. 

Some `.env` variables need to be set for this to work. 

### Type of change

- [ ] bug fix
- [ ] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

Try authenticating and making an announcement! 

Still TODO:

- [x] Clean up the CLIENT_SECRET configuration that is unneeded 
- [x] Implement the announcements endpoint on the CLI

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [x] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [x] Any non-clive `bindata.go` or `*.pb.go` file changed only has timestamps changed not actual data changed

### CODE REVIEW

**Review by pdeziel — APPROVED** [review]
The logic looks good to me, I wanted to test it but it doesn't look like I have the time today.

File: pkg/gds/emails/bindata.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func deliver_certsHtml() (*asset, error) {
 		return nil, err
 	}
 
-	info := bindataFileInfo{name: "deliver_certs.html", size: 1591, mode: os.FileMode(0644), modTime: time.Unix(1658243587, 0)}
+	info := bindataFileInfo{name: "deliver_certs.html", size: 1591, mode: os.FileMode(0644), modTime: time.Unix(1658373416, 0)}
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/gds/emails/bindata.go:100]: They did yes - normally I would just check them out and not commit this change; however I had to keep them due to a merge conflict with main (I branched this off of the email template PR).

File: pkg/bff/auth/clive/config.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		return errors.New("invalid configuration: missing $AUTH0_CLIENT_ID")
+	case c.TokenCache == "":
+		return errors.New("invalid configuration: missing path in $AUTH0_TOKEN_CACHE")
+	default:
+		return nil
+	}
+}
+
+func (c Config) IsZero() bool {
+	return c.Domain == "" && c.ClientID == "" && c.TokenCache == ""
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/bff/auth/clive/config.go:46]: Good catch; I'll add it!

**Comment by bbengfort** [comment]
> The logic looks good to me, I wanted to test it but it doesn't look like I have the time today.

Thanks! No worries about the testing - it's going to be Edwin and I using it so if it doesn't entirely work then we can deal with it then.

**Pull request merged**